### ISSUE TITLE
kubernetes.secret_present should allow `type` parameter

### ISSUE THREAD

**Issue by jodok** [saltstack/salt#46268]
### Description of Issue/Question

unfortunately the `type` parameter for a kubernetes secret can't be specified when using the `secret_present` state. therefor the type of the secret will always be `opaque`. but e.g. to specify an [ImagePullSecret](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod) you need the type `. dockerconfigjson`.

(the python client supports it: https://github.com/kubernetes-client/python/blob/master/kubernetes/client/models/v1_secret.py#L204)

### Versions Report
salt 2017.7.3

**Comment by Ch3LL** [comment]
looks like we are only passing data and metadata shown here: https://github.com/saltstack/salt/blob/v2017.7.3/salt/modules/kubernetes.py#L1073-L1075

seems a very reasonable thing to add. Want to give a go at a PR?

**Comment by hbokh** [comment]
This has not been added / fixed in `2018.3.3` nor `2019.2`, right?

https://github.com/saltstack/salt/blob/v2018.3/salt/modules/kubernetes.py#L1089-L1091

**Comment by Ch3LL** [comment]
yeah anyone else is welcome to push a PR for this :) would very much welcome that
